enhancements/local-storage/automatic-partition.md

```diff
@@ -0,0 +1,244 @@
+---
+title: automatic-partition-of-disks-in-Local-Storage-Operator
+authors:
+  - "@rohan47"
+reviewers:
+  - "@hekumar"
+  - "@sapillai"
+  - "@leseb"
+  - "@travisn"
+  - "@rohantmp"
+  - "@jarrpa"
+approvers:
+  - "@hekumar"
+creation-date: 2020-09-25
+last-updated: 2020-09-25
+status:
+---
+
+# Automatic partition of Disks in the Local-Storage-Operator
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This is a proposal to enable
+- automated partition of local devices via the `LocalVolumeSet` CR
+
+This CRD will be implemented in the LocalStorageOperator(LSO)
+
+## Motivation
+
+For performance reasons the user will want to use partitions.
+On NVMe devices, some of the devices are capable of delevering very high IOPS per device. An OSD tends to bottleneck long before it reaches anywhere close to the device IOPS limit.
+So with single OSD per NVMe device, there will be a big gap between measured or advertised capabilities of these devices and what you can get with Ceph/OCS on top of them.
+
+### Goals
+
+- Partition the selected devices based on the partition policy specified by the user.
+- If a LocalVolumeSet has partitioning policy, it will create partitions on all disks.
+
+### Non-Goals
+
+- If anything changes later such as the parition policy changes or the device is resized, we won't go add or change any partitions.
+- Policy for wiping device to clean the partitions
+- Mixed method of PV creation is not supported. i.e., using some devices entirely and some with partitoins in a single LocalVolumeSet is not supported.
+
+### Risks and Mitigations
+- It will depend entirely on LocalVolumeSet. The device filtering for LocalVolumeSet is sufficient for a device to qualify for partitioning. So if a disk has been verified to be used, we will use it.
+- An undisired device can be selected by LocalVolumeSet device discovery which qualifies the device selection criteria specified by the user.
+
+## Proposal
+
+The `local-storage-operator` is already capable of detecting local disks from the nodes and provisioning PVs out of them, but it can't partition them.
+The idea is to introduce the feature of partitioning the local disks and using the partitions to provision the PVs.
+
+The administrator can specify if he wants to partition the disks in the LocalVolumeSet CR and the diskmaker, after detecting the disk, LSO will partition them first and then provision PVs on those partitions.
```

Comment:
```suggestion
The administrator can specify if he wants to partition the disks in the LocalVolumeSet CR and the diskmaker, after detecting the disk, will partition them first before provisioning PVs on those partitions.
```

Reply:
New sentence for the diskmaker, otherwise it sounds like the admin can configure the diskmaker.